src/jimgw/single_event/runManager.py

```diff
@@ -190,23 +180,67 @@ def initialize_likelihood(self, prior: prior.Prior) -> SingleEventLiklihood:
             **self.run.data_parameters,
         )
 
-    def initialize_prior(self) -> prior.Prior:
+    def initialize_prior(self) -> prior.CombinePrior:
         priors = []
         for name, parameters in self.run.priors.items():
-            if parameters["name"] not in prior_presets:
-                raise ValueError(f"Prior {name} not recognized.")
-            if parameters["name"] == "EarthFrame":
-                priors.append(
-                    prior.EarthFrame(
-                        gps=self.run.data_parameters["trigger_time"],
-                        ifos=self.run.detectors,
+            assert isinstance(
+                parameters, dict
+            ), "Prior parameters must be a dictionary."
+            assert "name" in parameters, "Prior name must be provided."
+            assert isinstance(parameters["name"], str), "Prior name must be a string."
+            try:
+                prior_class = getattr(single_event_prior, parameters["name"])
+            except AttributeError:
+                try:
+                    prior_class = getattr(prior, parameters["name"])
+                except AttributeError:
+                    raise ValueError(f"{parameters['name']} not recognized.")
+            parameters.pop("name")
```

Comment:
Is there any specific reason you need to alter `self.run.priors` here? I don't think its a good idea to change anything in `SingleEventRun`, which is supposed to store static information of the run. This line is going to cause troubles if you want to reuse the object `SingleEventRun`.

Reply:
Solved in #140